src/modules/infobox/components/styling/FeatureStyleEdit.vue

```diff
@@ -32,15 +33,17 @@ const description = computed({
         return feature.description
     },
     set(value) {
-        store.dispatch('changeFeatureDescription', { feature, description: value })
+        this.description = value
+        debounce(store.dispatch('changeFeatureDescription', { feature, description: value }), 3000)
```

Comment:
The debounce helper is not working like this, it return a new function that would be debounced when calling, and after testing it effectively don't debounce.
You should do something like this
```javascript
const dispatch = debounce(store.dispatch, 500)
dispatch('changeFeatureDescription', { feature, description: value })
```
However I tried it quickly and it don't work well because the dispatch will overwrite what you has written in between the debouncing.
I think we should totally decoupled the local state from the store and use a normal ref. Then you can add a watch on the ref that would set the store (using eventually a debouncing only on the store). You can have a look at this PR/commit on how I solve the issue on the search bar https://github.com/geoadmin/web-mapviewer/pull/585/commits/4c4fe4618a02463cd65da87a1bb8d3fea8163d61

Reply:
hi @ltshb 
I can't use the same approach, since I need to compare if the text on the UI stays the same or not.
My approach is watching the UI state, whenever it changes, it will create a debounce to call the update function in the next 0.3 seconds. The function will compare the parameter that it receives with the current state of the UI (when the function runs). If they are the same (it means the user stops typing), it will update the value of the store.

I chose 0.3 seconds since 3 seconds is too long to see the text updated on the map. More than 1 second seems like a lag for me.

What do you think?